### ISSUE TITLE
Loss is nan when I used sdv2.1 to finetune

### ISSUE THREAD

**Issue by howardgriffin** [kohya-ss/sd-scripts#385]
Does anyone have the same problem, can you provide a example of train script?

**Comment by atYuguo** [comment]
Are you using Lion optimizer? I have this problem only with Lion.

**Comment by DKnight54** [comment]
I'm not sure if you experienced the same thing I was experiecing, where when I started training Loras, where it straight away went to loss=nan when training started. However, I managed to workaround by changing Mixed Precision to No. (Note, I'm using the GUI by [bmaltais](https://github.com/bmaltais/kohya_ss) which is usually a build or two behind kohya's sd-scripts.)

I'm using Adafactor as my optimizer, and have not used Lion, and I don't recall trying AdamW instead of AdamW8bit.
Using Xformers, Pytorch2 (Worked with the older original Pytorch as well, but main benefit was I was experiencing less hiccuping turning garbage collection and maybe slight improvement in training speeds).

Issue is? VRAM usage is up, and I'd like to go back to FP16 when training again, but have not being able to find any other setting that works.
![image](https://user-images.githubusercontent.com/126916963/230601292-308c6ea1-75ec-408b-a636-4f2257c13677.png)
![image](https://user-images.githubusercontent.com/126916963/230601352-7bda36dd-230a-4a47-bd1b-6b57c69bffc7.png)
Hopefully this helps, and if you find settings that allows FP16 precision, I'd love to hear it!

**Comment by DKnight54** [comment]
@howardgriffin Oh hey, was messing around with my settings again, and found out that setting mixed precision to BF16 worked for me. Perhaps you can try that out. Note that to my knowledge, this requires 30/40 series Nvidia GPU.

**Comment by howardgriffin** [comment]
@DKnight54 Thank you for your advice. I am using a T4 graphics card.  When I set the mixed precision to BF16, the error 'bf16 mixed precision requires PyTorch >= 1.10 and a supported device' occurred and my pytorch version is 1.13.1. 
When  I set the mixed precision to fp32, the loss seems normal. So, may it be related to the type of graphics card?

**Comment by DKnight54** [comment]
Sad to say, although loss was not NAN when I tried the bf16, the result was just noise for me. @kohya-ss do you have any advise? Do I need to reconfigure accelerate to bf16?
@howardgriffin I checked, and the T4 graphics card is using the turing architecture. Only the newer generations of cards using ampere or newer can support bf16.

**Comment by sdbds** [comment]
i use H100 and had same problem...
it is bad without Mixed Precision

**Comment by DKnight54** [comment]
@howardgriffin @sdbds I have not fully tested if this fixes using FP16 on all optimizers(I'm having my first success using AdamW optimizer, and suggest if you can try it yourself and see if it works for you too. It might be that something got fixed in the latest version of the SD-Scripts. 

Attaching my config from the GUI in txt form for reference.
[SD21_working.txt](https://github.com/kohya-ss/sd-scripts/files/11274589/SD21_working.txt)

Credit to @sashaok123 for pointing me in the correct direction.
